patchwork-events-world/src/main/java/net/patchworkmc/mixin/event/world/MixinMinecraftServer.java

```diff
@@ -47,42 +48,10 @@ public MixinMinecraftServer(String name) {
 	@Final
 	private Map<DimensionType, ServerWorld> worlds;
 
-	/*
-	// This is a variant of the world load hook that is less likely to break mods and more likely to break on updates.
-	// Should get called once per loop, regardless of which if branch it takes.
-	@Inject(
-		method = "createWorlds",
-		slice = @Slice(
-			from = @At(value = "INVOKE", target = "java/util/Iterator.hasNext ()Z")
-		),
-		at = @At(value = "JUMP", opcode = Opcodes.GOTO),
-		locals = LocalCapture.CAPTURE_FAILHARD
-	)
-	private void hookCreateWorlds(WorldSaveHandler worldSaveHandler, LevelProperties properties, LevelInfo levelInfo, WorldGenerationProgressListener worldGenerationProgressListener, CallbackInfo ci, ServerWorld serverWorld, ServerWorld serverWorld2, Iterator var7, DimensionType dimensionType) {
-		WorldEvents.onWorldLoad(this.worlds.get(dimensionType));
-	}
-
-	*/
-
-	// This injection gets called at the beginning of each loop, and is used to special case the overworld dimension type.
-	@Redirect(method = "createWorlds", at = @At(value = "INVOKE", target = "java/util/Iterator.next ()Ljava/lang/Object;"))
-	private Object proxyNextWorldToSpecialCaseOverworld(Iterator<DimensionType> iterator) {
-		DimensionType type = iterator.next();
-
-		if (type == DimensionType.OVERWORLD) {
-			WorldEvents.onWorldLoad(this.worlds.get(type));
-		}
-
-		return type;
-	}
-
-	// This injection handles every other dimension type.
-	@Redirect(method = "createWorlds", at = @At(value = "INVOKE", target = "java/util/Map.put (Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;", ordinal = 1))
-	private Object proxyPutWorld(Map<Object, Object> worlds, Object type, Object world) {
-		worlds.put(type, world);
-		WorldEvents.onWorldLoad((ServerWorld) world);
-
-		return world;
+	// Fabric usually fires the event much earlier in the method, so this is just picking a point closer to when Forge would fire it.
+	@Inject(method = "createWorlds", at = @At(value = "INVOKE", target = "net/minecraft/world/dimension/DimensionType.getAll ()Ljava/lang/Iterable;"))
```

Comment:
Shouldn't the inject target be in `Lnet/minecraft/whatever` format?

Reply:
hm. good question, and i hope i haven't goofed my previous PRs because of this. I've just been copying from IDEA's bytecode viewer: `INVOKESTATIC net/minecraft/world/dimension/DimensionType.getAll ()Ljava/lang/Iterable;`

Reply:
Well, it at least seems to work as intended as-is. I'll leave "should it be changed anyway" as something for future consideration.

Reply:
i con't care about the signature format as long as it's readable and it applies